src/eureka/S4cal_StellarSpectra/s4cal_StellarSpec.py

```diff
@@ -0,0 +1,285 @@
+                                            meta.base_dur))[0][0]
+            except IndexError:
+                it0 = 0
+            try:
+                it5 = np.where(spec.time > (t0 + meta.t14/2 +
+                                            meta.base_dur))[0][0]
+            except IndexError:
+                it5 = -1
+        meta.it = [it0, it1, it2, it3, it4, it5]
+
+        if meta.isplots_S4cal >= 2:
+            fig, ax = plot_whitelc(optspec, spec.time, meta, i, fig=fig, ax=ax)
+
+        # Median baseline (i.e. out-of-eclipse) flux
+        spec_baseline = np.ma.median(np.ma.concatenate(
+            (optspec[it0:it1], optspec[it4:it5])), axis=0)
+        std_baseline = np.ma.std(np.ma.concatenate(
+            (optspec[it0:it1], optspec[it4:it5])), axis=0)
+        # Mask outliers along wavelength axis
+        tck = spi.splrep(wave, spec_baseline, w=1/std_baseline, k=3)
+        spline = spi.splev(wave, tck)
+        mask = sigrej(spec_baseline - spline, meta.sigma_thresh)
+        igood = np.where(~mask)[0]
+        # Create XArray data arrays
+        base_flux = xrio.makeLCDA(spec_baseline[igood, np.newaxis],
+                                  wave[igood], [t0],
+                                  spec.optspec.flux_units,
+                                  spec.optspec.wave_units,
+                                  spec.optspec.time_units,
+                                  name='Median Baseline Flux')
+        base_fstd = xrio.makeLCDA(std_baseline[igood, np.newaxis],
+                                  wave[igood], [t0],
+                                  spec.optspec.flux_units,
+                                  spec.optspec.wave_units,
+                                  spec.optspec.time_units,
+                                  name='Std. Dev. of Baseline Flux')
+
+        # Median in-eclipse (or in-transit) flux
+        spec_t23 = np.ma.median(optspec[it2:it3+1], axis=0)
+        std_t23 = np.ma.std(optspec[it2:it3+1], axis=0)
+        # Mask outliers along wavelength axis
+        tck = spi.splrep(wave, spec_t23, w=1/std_t23, k=3)
+        spline = spi.splev(wave, tck)
+        mask = sigrej(spec_t23 - spline, meta.sigma_thresh)
+        igood = np.where(~mask)[0]
+        # Create XArray data arrays
+        ecl_flux = xrio.makeLCDA(spec_t23[igood, np.newaxis],
+                                 wave[igood], [t0],
+                                 spec.optspec.flux_units,
+                                 spec.optspec.wave_units,
+                                 spec.optspec.time_units,
+                                 name='Median In-Eclipse Flux')
+        ecl_fstd = xrio.makeLCDA(std_t23[igood, np.newaxis],
+                                 wave[igood], [t0],
+                                 spec.optspec.flux_units,
+                                 spec.optspec.wave_units,
+                                 spec.optspec.time_units,
+                                 name='Std. Dev. of In-Eclipse Flux')
```

Comment:
Much of this code should be broken out into smaller, task-specific functions to maintain modularity and ease of code review